.github/workflows/monorepo-semantic-release.yaml

```diff
@@ -0,0 +1,63 @@
+name: Release
+
+on:
+  workflow_call:
+    inputs:
+      directory:
+        description: "The directory containing module directories which should be individually released. e.g. 'modules/'"
+        required: true
+        type: string
+      suffix:
+        description: "An optional suffix for determining which paths of a module consitute a releasable change. Defaults to '/**' denoting any files within a module."
+        required: false
+        type: string
+        default: "/**"
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.matrix.outputs.value }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Find all module names/locations
+        # Outputs a files.yaml for tj-actions/changed-files with keys of all files/directories in the input and search paths matching the given directory path plus an optional suffix. i.e.
+        # module_one:
+        #   - modules/module_one/**
+        # module_two:
+        #   - modules/module_two/**
+        run: >
+          ls -d ${{ inputs.directory }}/*/ |
+          xargs -n 1 basename |
+          jq -R -s -c 'split("\n")[:-1]' |
+          yq -p=json '.[] as $i ireduce ({}; .[$i] = ["${{ inputs.directory }}"+ $i + "${{ inputs.suffix }}"])' > .changed_files_config.yaml &&
```

Comment:
Personally I like explicitly installing tools that aren't in a standard os, but maybe that's an "old man yells at github actions". I like arkade for this. Feel free to ignore

Reply:
I don't feel strongly about this, so I'm going to leave as is. But I sympathize with the old man sensibilities.